app/helpers/check_resolvability_helper.rb

```diff
@@ -49,31 +49,37 @@ def follow_redirection(url, format, timeout_seconds, redirect_limit = resolvabil
     redirect_count = 0
     redirections = [uri]
 
-    until (!response.nil? && !response.is_a?(Net::HTTPRedirection)) || redirect_count >= redirect_limit
-      http = Net::HTTP.new(uri.host, uri.port)
-      http.use_ssl = (uri.scheme == 'https')
-      http.open_timeout = timeout_seconds
-      begin
-        response = Timeout.timeout(timeout_seconds) { http.request_head(uri, 'Accept' => format) }
-      rescue Timeout::Error, Net::OpenTimeout
-        return resolvability_status('Timeout', [], redirections, result: 0)
-      end
-
-      if response.is_a?(Net::HTTPRedirection) && response['location']
-        uri = URI.join(uri, response['location'])
-        redirections << uri
-        redirect_count += 1
+    total_time = Benchmark.measure do
+      until (!response.nil? && !response.is_a?(Net::HTTPRedirection)) || redirect_count >= redirect_limit
+        http = Net::HTTP.new(uri.host, uri.port)
+        http.use_ssl = (uri.scheme == 'https')
+        http.open_timeout = timeout_seconds
+        begin
+          response = Timeout.timeout(timeout_seconds) { http.request_head(uri, 'Accept' => format) }
+        rescue Timeout::Error, Net::OpenTimeout
+          return resolvability_status('Timeout', [], redirections, result: 0, response_time: total_time.real.round(3))
```

Comment:
this will not wokr as `total_time` variable is not available inside this block. Only available after the block.